src/main/java/org/broadinstitute/hellbender/utils/pileup/ReadPileup.java

```diff
@@ -44,7 +44,9 @@ public ReadPileup(final Locatable loc, final List<PileupElement> pileup) {
      * Note: the current implementation of ReadPileup does not efficiently retrieve the stratified pileup
      */
     public ReadPileup(final Locatable loc, final Map<String, ReadPileup> stratifiedPileup) {
-        this(loc, stratifiedPileup.values().stream().flatMap(ReadPileup::getElementStream).collect(Collectors.toList()));
+        this(loc, stratifiedPileup.values().stream().flatMap(ReadPileup::getElementStream)
+                .sorted((l, r) -> Integer.compare(l.getRead().getStart(), r.getRead().getStart()))
+                .collect(Collectors.toList()));
```

Comment:
The point is that if it is already sorted, we are paying the cost of traversal in `FragmentCollection.create()`. Nevertheless, I'm working in a better way of handling the list of pileup elements (#2309).

I think that in the meanwhile will be better create a method for traversal of pileup elements in sort order, and use that one in the `FragmentCollection.create()` method.